File: algorithms/2586.count-the-number-of-vowel-strings-in-range/solution.go
func vowelStrings(words []string, left int, right int) int {
	res := 0
	for _, word := range words[left : right+1] {
		if isVowel(word[0]) && isVowel(word[len(word)-1]) {
			res++
		}
	}
	return res
}

func isVowel(b byte) bool {
	for _, v := range []byte("aeiou") {
		if v == b {
			return true
		}
	}
	return false
}

File: algorithms/2586.count-the-number-of-vowel-strings-in-range/solution.rs
impl Solution {
    pub fn vowel_strings(words: Vec<String>, left: i32, right: i32) -> i32 {
        let vowels: Vec<char> = vec!['a', 'e', 'i', 'o', 'u'];
        let mut res: i32 = 0;
        for word in &words[(left as usize)..=(right as usize)] {
            let chars: Vec<char> = word.chars().collect();
            if vowels.contains(&chars[0]) && vowels.contains(&chars[chars.len() - 1]) {
                res += 1;
            }
        }
        res
    }
}
